src/plugins/dbus/receivemessage.c

```diff
@@ -8,8 +8,91 @@
 
 #include "dbus.h"
 
+#include <stdio.h>
+
+#define RECEIVE_MATCH_RULE "type='signal',interface='org.libelektra',path='/org/libelektra/configuration'"
+
+/**
+ * @internal
+ * Setup D-Bus connection for receiving Elektra's signal messages.
+ *
+ * @param  connection  D-Bus connection
+ * @param  filter_func message handler
+ * @param  data        data passed to message handler
+ * @retval 1 on success
+ * @retval 0 on error
+ */
+int elektraDbusSetupReceiveMessage (DBusConnection * connection, DBusHandleMessageFunction filter_func, void * data)
+{
+	ELEKTRA_NOT_NULL (connection);
+	ELEKTRA_NOT_NULL (filter_func);
+
+	DBusError error;
+	dbus_error_init (&error);
+
+	dbus_bus_add_match (connection, RECEIVE_MATCH_RULE, &error);
+	if (dbus_error_is_set (&error)) goto error;
+
+	if (!dbus_connection_add_filter (connection, filter_func, data, NULL))
+	{
+		goto error;
+	}
+
+	dbus_error_free (&error);
+	return 1;
+
+error:
+	printf ("Error occurred\n");
+	dbus_error_free (&error);
+	return 0;
+}
+
+/**
+ * @internal
+ * Revert changes made to D-Bus connection by elektraDbusSetupReceiveMessage().
+ *
+ * @param  connection  D-Bus connection
+ * @param  filter_func message handler
+ * @param  data        data passed to message handler
+ * @retval 1 on success
+ * @retval 0 on error
+ */
+int elektraDbusTeardownReceiveMessage (DBusConnection * connection, DBusHandleMessageFunction filter_func, void * data)
+{
+	ELEKTRA_NOT_NULL (connection);
+	ELEKTRA_NOT_NULL (filter_func);
+
+	DBusError error;
+	dbus_error_init (&error);
+
+	dbus_bus_remove_match (connection, RECEIVE_MATCH_RULE, &error);
+	if (dbus_error_is_set (&error)) goto error;
+
+	dbus_connection_remove_filter (connection, filter_func, data);
+
+	dbus_error_free (&error);
+	return 1;
+
+error:
+	printf ("Error occurred\n");
```

Comment:
Is this library or test code?

Reply:
It is test code. I added this info to the file header and excluded the file from the plugin sources.